File: Dao/pedidos-rest-24-main/pedidos-rest-24-main/src/main/java/es/uvigo/mei/pedidos/servicios/ClienteService.java
package es.uvigo.mei.pedidos.servicios;

import java.util.List;
import java.util.Optional;

import es.uvigo.mei.pedidos.entidades.Cliente;

public interface ClienteService {
	public Cliente crear(Cliente cliente);
	public Cliente modificar(Cliente cliente);
	public void eliminar(Cliente cliente);
	public Optional<Cliente> buscarPorDNI(String dni);
	public List<Cliente> buscarTodos();
	public List<Cliente> buscarPorNombre(String patron);
	public List<Cliente> buscarPorLocalidad(String localidad);
}

File: Dao/pedidos-react-23-main/pedidos-react-23-main/src/services/clienteService.js
import api from "./pedidosAPI";

class ClienteService {
  buscarTodos() {
    return api.get("/clientes");
  }

  buscarPorDNI(dni) {
    return api.get(`/clientes/${dni}`);
  }

  crear(data) {
    return api.post("/clientes", data);
  }

  modificar(dni, data) {
    return api.put(`/clientes/${dni}`, data);
  }

  eliminar(dni) {
    return api.delete(`/clientes/${dni}`);
  }

  buscarPorNombre(nombre) {
    return api.get(`/clientes?nombre=${nombre}`);
  }

  buscarPorLocalidad(localidad) {
    return api.get(`/clientes?localidad=${localidad}`);
  }
}

const service = new ClienteService();
export default service;
